.github/workflows/tests-ci.yml

```diff
@@ -70,65 +66,96 @@ jobs:
+        uses: actions/checkout@v4
 
       - name: Set up Node.js
-        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        uses: actions/setup-node@v4
         with:
           node-version: 18
           registry-url: https://registry.npmjs.org/
-
-      - name: Install latest npm
-        run: npm install -g npm@latest
+          cache: "npm"
 
       - name: Install dependencies
         run: npm ci
 
-      - name: Install Playwright Browsers
+      - name: Get Playwright Version (for cache)
+        id: get-playwright-version
+        run: |
+          PLAYWRIGHT_VERSION=$(npm view @playwright/test version)
+          echo "Playwright Version: $PLAYWRIGHT_VERSION"
+          echo "PLAYWRIGHT_VERSION=$PLAYWRIGHT_VERSION" >> $GITHUB_ENV
+
+      - name: Restore Cached Playwright Browsers
+        id: cache-playwright-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ~/Library/Caches/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ env.PLAYWRIGHT_VERSION }}
+
+      - name: Install Playwright Browsers (if no cache)
+        if: steps.cache-playwright-restore.outputs.cache-hit != 'true'
         run: npx playwright install --with-deps
 
-      - name: Build all workspace packages
-        run: npm run build
+      - name: Save Playwright Cache (if no cache)
+        uses: actions/cache/save@v3
+        id: cache-playwright-save
+        if: steps.cache-playwright-restore.outputs.cache-hit != 'true'
+        with:
+          path: ~/Library/Caches/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ env.PLAYWRIGHT_VERSION }}
 
-      - name: Install docker
-        run: brew install docker && brew install docker-compose # && colima start
+      - name: Build esm
+        run: npm run build:esm --ws
 
-      - name: Start docker
-        run: colima start
+      - name: Build browser for matrix ${{ matrix.group }}
+        run: npm run build:browser ${{ matrix.packages }}
 
-      - name: Start dwn-server container
-        run: cd packages/dev-env && docker-compose up -d
+      - name: Run dwn-server (background)
+        run: |
+          node node_modules/@web5/dwn-server/dist/esm/src/main.js &
+          echo "DWN_SERVER_BACKGROUND_PROCESS=$!" >> $GITHUB_ENV
 
-      - name: Wait for dwn-server to be ready
-        run: until curl -sf http://localhost:3000/health; do echo -n .; sleep .1; done
+      - name: Run tests for matrix ${{ matrix.group }}
+        run: npm run test:browser ${{ matrix.packages }}
 
-      - name: Run tests for all packages
-        run: npm run test:browser --ws
+      - name: Terminate dwn-server
+        run: kill $DWN_SERVER_BACKGROUND_PROCESS || true
 
   web5-spec:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        uses: actions/checkout@v4
```

Comment:
[Security scanning](https://github.com/TBD54566975/web5-js/security/code-scanning) checks for pinned dependencies using hash so pin to a v4 using the commit hash of the GitHub action.

> For GitHub workflows used in building and releasing your project, pin dependencies by hash.